maven-repository-manager/src/main/java/org/jboss/pnc/mavenrepositorymanager/RepositoryManagerDriver.java

```diff
@@ -270,14 +276,27 @@ private void setupBuildRepos(BuildExecution execution, Indy indy)
             // Global-level repos, for captured/shared artifacts and access to the outside world
             addGlobalConstituents(buildGroup, tempBuild);
 
-            // add extra repositories removed from poms by the adjust process
-            addExtraConstituents(execution.getArtifactRepositories(), id, buildContentId, indy, buildGroup);
+            // add extra repositories removed from poms by the adjust process and set in BC by user
+            List<ArtifactRepository> extraDependencyRepositories = extractExtraRepositoriesFromGenericParameters(genericParameters);
+            extraDependencyRepositories.addAll(execution.getArtifactRepositories());
+            addExtraConstituents(extraDependencyRepositories, id, buildContentId, indy, buildGroup);
 
             indy.stores().create(buildGroup, "Creating repository group for resolving artifacts in build: " + id
                     + " (repo: " + buildContentId + ")", Group.class);
         }
     }
 
+    private List<ArtifactRepository> extractExtraRepositoriesFromGenericParameters(Map<String, String> genericParameters) {
+        String extraReposString = genericParameters.get(EXTRA_PUBLIC_REPOSITORIES_KEY);
+        if (extraReposString == null)
+            return Collections.emptyList();
```

Comment:
Just noticed that this will cause issues on line 261 - EmptyList class does not support add and addAll methods, it throws UnsupportedOperationException when you try to use them. So you should return something like "new ArrayList<>()" instead.

Reply:
You are commenting on an outdated version ;-). It was fixed in the next commits as it was discovered by failing test ;-)